implementations/rust/ockam/ockam/src/entity/profile/profile_change_type/create_key.rs

```diff
@@ -0,0 +1,100 @@
+use crate::entity::profile::event_identifier::EventIdentifier;
+use crate::entity::profile::profile_change::{ProfileChange, ProfileKeyPurpose, ProfileKeyType};
+use crate::entity::profile::profile_change_type::ProfileChangeType;
+use crate::entity::profile::proof::{ProfileChangeProof, Signature, SignatureType};
+use crate::entity::{Changes, Profile, ProfileChangeEvent, ProfileEventAttributes};
+use crate::OckamError;
+use ockam_vault_core::types::{SecretAttributes, SecretPersistence, SecretType};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct CreateKeyChange {
+    label: String,
+    key_type: ProfileKeyType,
+    key_purpose: ProfileKeyPurpose,
+    public_key: Vec<u8>,
+}
+
+impl CreateKeyChange {
+    pub fn new(
+        label: String,
+        key_type: ProfileKeyType,
+        key_purpose: ProfileKeyPurpose,
+        public_key: Vec<u8>,
+    ) -> Self {
+        CreateKeyChange {
+            label,
+            key_type,
+            key_purpose,
+            public_key,
+        }
+    }
+}
+
+impl CreateKeyChange {
+    pub fn label(&self) -> &str {
+        &self.label
+    }
+    pub fn key_type(&self) -> ProfileKeyType {
+        self.key_type
+    }
+    pub fn key_purpose(&self) -> ProfileKeyPurpose {
+        self.key_purpose
+    }
+    pub fn public_key(&self) -> &[u8] {
+        &self.public_key
+    }
+}
+
+impl Profile {
+    pub(crate) fn create_key_event(
```

Comment:
Looks similar to create_profile. Should refactor when you can.

Reply:
Agree